### PR TITLE
Optimal extraction

### PR DESCRIPTION
**Pull request by ajwheeler**

### CODE REVIEW

File: pipeline.jl [hunk 1]
```diff
@@ -177,6 +177,7 @@ git_branch, git_commit = initalize_git(src_dir);
 
         # extraction 3D -> 2D
         dimage, ivarimage, chisqimage, CRimage = if extractMethod_loc == "dcs"
+            # TODO some kind of outlier rejection, this just keeps all diffs
```
Comment (ajwheeler) [inline pipeline.jl:180]: not strictly relevant here, just noting that it needs to be done

File: pipeline_2d_1d.jl [hunk 2]
```diff
@@ -117,16 +124,25 @@ git_branch, git_commit = initalize_git(src_dir);
...
+        else # "optimal"
+            flux_1d, ivar_1d, mask_1d = extract_optimal(
+                dimage, ivarimage, pix_bitmask, regularized_trace_params)
+        end
 
         # we probably want to append info from the fiber dictionary from alamanac into the file name
         # outfname = replace(fname, "ap2Dcal" => "ap1D")
         outfname = replace(fname, "ap2D" => "ap1D")
-        jldsave(outfname; flux_1d, ivar_1d, mask_1d, git_branch, git_commit) #ivarimage, pix_bitmask come back and put analogies of these in
+        jldsave(outfname; flux_1d, ivar_1d, mask_1d, git_branch, git_commit) # ivarimage, pix_bitmask come back and put analogies of these in
```
Comment (ajwheeler) [inline pipeline_2d_1d.jl:145]: Is this comment outdated?

**Comment by ajwheeler** [comment]
TODO
- understand if ivar should be \propto flux, because it is
- pull `process` files into package proper, test them.
- refactor so that boxcar and optimal are parallel
- put some plots here
- add some plots to the slack bot?

**Comment by andrew-saydjari** [comment]
Yeah, I don't have a good intuition for what happens to ivars after being transformed in the gaussian fitting. But before that, I really believe that flux.*ivar should be constant (and related/be the gain). So anything that breaks that is something I want to grok before we merge it.

**Comment by ajwheeler** [comment]
Turns out (at least for the data I'm looking at), this makes a very tiny difference.

image
image

On the bright side, it looks like the flux * ivar is behaving as expected for for optimal and boxcar extraction.
image

**Review by andrew-saydjari — COMMENTED** [review]
looks good to me

**Pull request merged**